File: app/src/test/java/dev/bee/kanjianki/SettingsStudyAheadPanelModelsTest.java
package dev.bee.kanjianki;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public final class SettingsStudyAheadPanelModelsTest {
    @Test
    public void createBuildsStudyAheadPanelModelFromCurrentMinutes() {
        RecordingSaveAction action = new RecordingSaveAction();

        SettingsStudyAheadPanelModel model = SettingsStudyAheadPanelModels.create(30, action);

        assertEquals("Study ahead", model.getTitle());
        assertEquals("Minutes (0-1440)", model.getMinutesLabel());
        assertEquals("30", model.getInitialMinutesText());
        assertEquals("Save study ahead", model.getSaveLabel());

        model.getOnSave().save("45");
        assertEquals("45", action.minutesText);
    }

    private static final class RecordingSaveAction implements SettingsStudyAheadSaver {
        String minutesText;

        @Override
        public void save(String minutesText) {
            this.minutesText = minutesText;
        }
    }
}

File: app/src/test/kotlin/dev/bee/kanjianki/SettingsStudyAheadPanelModelsTest.kt
package dev.bee.kanjianki

import org.junit.Assert.assertEquals
import org.junit.Test

class SettingsStudyAheadPanelModelsTest {
    @Test
    fun createBuildsStudyAheadPanelModelFromCurrentMinutes() {
        var minutesText: String? = null

        val model = SettingsStudyAheadPanelModels.create(30) { minutesText = it }

        assertEquals("Study ahead", model.title)
        assertEquals("Review early; learning waits stay fixed. Early answers never move the ladder.", model.body)
        assertEquals("Minutes (0-1440)", model.minutesLabel)
        assertEquals("30", model.initialMinutesText)
        assertEquals("Save minutes", model.saveLabel)

        model.onSave.save("45")
        assertEquals("45", minutesText)
    }
}
